test/v1/controllers/RetrieveOtherExpensesControllerSpec.scala

```diff
@@ -0,0 +1,155 @@
+    Some(PatentRoyaltiesPayments(Some("ROYALTIES PAYMENTS"), 2000.10))
+  )
+
+  "handleRequest" should {
+    "return Ok" when {
+      "the request received is valid" in new Test {
+
+        MockRetrieveOtherExpensesRequestParser
+          .parse(rawData)
+          .returns(Right(requestData))
+
+        MockRetrieveOtherExpensesService
+          .retrieveOtherExpenses(requestData)
+          .returns(Future.successful(Right(ResponseWrapper(correlationId, responseBody))))
+
+        MockHateoasFactory
+          .wrap(responseBody, RetrieveOtherExpensesHateoasData(nino, taxYear))
+          .returns(HateoasWrapper(responseBody, Seq(testHateoasLink)))
+
+        val result: Future[Result] = controller.handleRequest(nino, taxYear)(fakeRequest)
+        status(result) shouldBe OK
+        header("X-CorrelationId", result) shouldBe Some(correlationId)
+      }
+    }
+    "return the error as per spec" when {
+      "parser errors occur" should {
+        def errorsFromParserTester(error: MtdError, expectedStatus: Int): Unit = {
+          s"a ${error.code} error is returned from the parser" in new Test {
+
+            MockRetrieveOtherExpensesRequestParser
+              .parse(rawData)
+              .returns(Left(ErrorWrapper(Some(correlationId), error, None)))
+
+            val result: Future[Result] = controller.handleRequest(nino, taxYear)(fakeRequest)
+
+            status(result) shouldBe expectedStatus
+            contentAsJson(result) shouldBe Json.toJson(error)
+            header("X-CorrelationId", result) shouldBe Some(correlationId)
+          }
+        }
+
+        val input = Seq(
+          (BadRequestError, BAD_REQUEST),
+          (NinoFormatError, BAD_REQUEST),
+          (TaxYearFormatError, BAD_REQUEST),
+          (RuleTaxYearRangeInvalidError, BAD_REQUEST)
+        )
+
+        input.foreach(args => (errorsFromParserTester _).tupled(args))
+      }
+
+      "service errors occur" should {
+        def serviceErrors(mtdError: MtdError, expectedStatus: Int): Unit = {
+          s"a $mtdError error is returned from the service" in new Test {
+
+            MockRetrieveOtherExpensesRequestParser
+              .parse(rawData)
+              .returns(Right(requestData))
+
+            MockRetrieveOtherExpensesService
+              .retrieveOtherExpenses(requestData)
+              .returns(Future.successful(Left(ErrorWrapper(Some(correlationId), mtdError))))
+
+            val result: Future[Result] = controller.handleRequest(nino, taxYear)(fakeRequest)
+
+            status(result) shouldBe expectedStatus
+            contentAsJson(result) shouldBe Json.toJson(mtdError)
+            header("X-CorrelationId", result) shouldBe Some(correlationId)
+          }
+        }
+
+        val input = Seq(
+          (NinoFormatError, BAD_REQUEST),
+          (TaxYearFormatError, BAD_REQUEST),
+          (RuleTaxYearRangeInvalidError, BAD_REQUEST),
```

Comment:
don't need this error from the service

Reply:
Those set of errors aren't des errors, those are backend errors so RuleTaxYearRangeInvalid would be needed here